src/lib/viewers/media/DashViewer.js

```diff
@@ -505,6 +505,18 @@ class DashViewer extends VideoBaseViewer {
                 this.handleDownloadError(error, downloadURL);
                 return;
             }
+
+            if (
+                normalizedShakaError.code === shaka.util.Error.Code.BAD_HTTP_STATUS &&
+                normalizedShakaError.data[1] === 401 // token expired
+            ) {
+                this.refreshToken().then(newToken => {
```

Comment:
Does this handle the case where there is no tokenGenerator function? Should we add a exit condition so we don't retry infinitely?